app/src/main/java/com/example/dinhh/soundscape/device/Library.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.dinhh.soundscape.device
+
+import android.util.Log
+import com.example.dinhh.soundscape.data.Model
+import com.example.dinhh.soundscape.data.remote.SoundscapeApi
+import com.example.dinhh.soundscape.presentation.ListItem
+import io.reactivex.Completable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import java.io.IOException
+
+interface Library {
+    fun beginSearch(selectedCategory: String): Completable
+}
+
+class LibraryImpl: Library{
+    override fun beginSearch(selectedCategory: String): Completable {
+        return Completable.create {
+            Model.sounds.clear()
+            val apiKey =
+                "jFBaDxPcNzYZGu-gNMZ2L9-TjP1JjWl8OHFhdJV54gL82_M0cZi8oGEg-fB7gw3EpYvN0IHrHFP-Ic5sULo-iAWTl0k_y0t3CwrCQPpbYJkVIjmCV1Zzo0NB52ZLanwN"
+            try {
+                disposable =
+                        wikiApiServe.getSounds(
+                            key = apiKey,
+                            category = selectedCategory
+                        )
+                            .subscribeOn(Schedulers.io())
+                            .observeOn(AndroidSchedulers.mainThread())
```

Comment:
subscribeOn and observeOn is not necessary here because you use it in the UseCase